browser/caliban.py

```diff
@@ -1001,13 +1015,27 @@ def action_watershed(self, label, frame, x1_location, y1_location, x2_location,
         # apply watershed transform to the subsections
         ws = watershed(-img_sub_raw_scaled, img_sub_seeds, mask=img_sub_ann.astype(bool))
 
+        # did watershed effectively create a new label?
+        new_pixels = np.count_nonzero(np.logical_and(ws == new_label, img_sub_ann == current_label))
+        # if only a few pixels split, dilate them; new label is "brightest"
+        # so will expand over other labels and increase area
+        if new_pixels < 5:
+            ws = dilation(ws, disk(3))
+
+        # ws may only leave a few pixels of old label
+        old_pixels = np.count_nonzero(ws == current_label)
+        if old_pixels < 5:
+            # create dilation image so "dimmer" label is not eroded by "brighter" label
+            dilated_ws = dilation(np.where(ws==current_label, ws, 0), disk(3))
+            ws = np.where(dilated_ws==current_label, dilated_ws, ws)
+
```

Comment:
Any way to easily pull this out of the class and into a Util for DRY purposes? Or should that wait for a future re-factor?

Reply:
I would really like to refactor the actions for DRY purposes but I think that's a future step.